src/main/java/core/basesyntax/strategyimpl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategyimpl;
+
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.serviceimpl.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+
+    @Override
+    public OperationHandler get(FruitTransaction.Operation operation) {
+        Map<FruitTransaction.Operation, OperationHandler> strategy = new HashMap<>();
+        strategy.put(FruitTransaction.Operation.BALANCE, new BalanceHandler());
+        strategy.put(FruitTransaction.Operation.PURCHASE, new PurchaseHandler());
+        strategy.put(FruitTransaction.Operation.RETURN, new ReturnHandler());
+        strategy.put(FruitTransaction.Operation.SUPPLY, new SupplyHandler());
```

Comment:
This is very inefficient, it should be a class field, initialized in main and passed here in the constructor. 
Or if you want you can remove this class, and use this map directly in the `TransactionExecutor`, but still with initializing it in main